dali/test/python/operator_1/test_clahe.py

```diff
@@ -0,0 +1,242 @@
+    """Test CLAHE with RGB images on GPU."""
+    batch_size = 2
+    pipe = ClahePipeline(
+        "gpu",
+        batch_size,
+        input_shape=(64, 64, 3),
+        tiles_x=4,
+        tiles_y=4,
+        clip_limit=3.0,
+        luma_only=True,
+    )
+    pipe.build()
+
+    outputs = pipe.run()
+    input_data, clahe_output = outputs
+
+    # Verify output properties
+    assert len(clahe_output) == batch_size
+    for i in range(batch_size):
+        original = np.array(input_data[i])
+        enhanced = np.array(clahe_output[i])
+
+        assert original.shape == enhanced.shape == (64, 64, 3)
+        assert original.dtype == enhanced.dtype == np.uint8
+        assert 0 <= enhanced.min() and enhanced.max() <= 255
+
+
+def test_clahe_ops_api():
+    """Test CLAHE using the ops API."""
+    batch_size = 2
+    pipe = ClaheOpsPipeline(
+        "gpu", batch_size, input_shape=(32, 32, 1), tiles_x=2, tiles_y=2, clip_limit=1.5
+    )
+    pipe.build()
+
+    outputs = pipe.run()
+    input_data, clahe_output = outputs
+
+    # Verify output properties
+    assert len(clahe_output) == batch_size
+    for i in range(batch_size):
+        original = np.array(input_data[i])
+        enhanced = np.array(clahe_output[i])
+
+        assert original.shape == enhanced.shape == (32, 32, 1)
+        assert original.dtype == enhanced.dtype == np.uint8
+
+
+def test_clahe_parameter_validation():
+    """Test parameter validation for CLAHE operator."""
+    batch_size = 1
+
+    # Valid parameters should work
+    pipe = ClahePipeline("gpu", batch_size, tiles_x=8, tiles_y=8, clip_limit=2.0)
+    pipe.build()
+
+    # Test with different valid parameter combinations
+    valid_configs = [
+        {"tiles_x": 4, "tiles_y": 4, "clip_limit": 1.5},
+        {"tiles_x": 16, "tiles_y": 8, "clip_limit": 3.0},
+        {"tiles_x": 2, "tiles_y": 2, "clip_limit": 1.0},
+    ]
+
+    for config in valid_configs:
+        pipe = ClahePipeline("gpu", batch_size, **config)
+        pipe.build()
+        outputs = pipe.run()
+        assert len(outputs[1]) == batch_size
+
+
+def test_clahe_different_tile_configurations():
+    """Test CLAHE with different tile configurations."""
+    batch_size = 2
+
+    # Test different tile configurations
+    tile_configs = [
+        (2, 2),  # Few tiles
+        (4, 4),  # Standard
+        (8, 8),  # Many tiles
+        (4, 8),  # Asymmetric
+    ]
+
+    for tiles_x, tiles_y in tile_configs:
+        pipe = ClahePipeline(
+            "gpu",
+            batch_size,
+            input_shape=(64, 64, 1),
+            tiles_x=tiles_x,
+            tiles_y=tiles_y,
+            clip_limit=2.0,
+        )
+        pipe.build()
+
+        outputs = pipe.run()
+        input_data, clahe_output = outputs
+
+        # Verify all outputs are valid
+        for i in range(batch_size):
+            enhanced = np.array(clahe_output[i])
+            assert enhanced.shape == (64, 64, 1)
+            assert enhanced.dtype == np.uint8
```

Comment:
The tests only check the shape and type.
How about comparing against some reference implementation such as OpenCV, to check correctness?